src/interface.cpp

```diff
@@ -59,3 +60,41 @@ Rcpp::List aum_sort_interface
      Rcpp::Named("derivative_mat", out_deriv_mat)
      ) ;
 }
+
+// [[Rcpp::export]]
+Rcpp::List aumLineSearch(const Rcpp::DataFrame df, const double initialAum, int maxIterations) {
+    // extract columnds from dataframe
+    Rcpp::NumericVector fpDiff = df["fp.diff"];
+    Rcpp::NumericVector fnDiff = df["fn.diff"];
+    Rcpp::NumericVector intercept = df["intercept"];
+    Rcpp::NumericVector slope = df["slope"];
+    int lineCount = df.nrow();
+
+    // build lines
+    std::vector<Line> lines;
+    lines.reserve(lineCount);
+    for (int i = 0; i < lineCount; i++) {
+        Line line = Line { slope[i], intercept[i] };
+        lines.push_back(line);
+    }
+
+    Rcpp::NumericVector FP(lineCount, 0.0);
+    Rcpp::NumericVector FN(lineCount, 0.0);
+    Rcpp::NumericVector M(lineCount, 0.0);
+    Rcpp::NumericVector stepSizeVec(maxIterations, -1.0);
+    Rcpp::NumericVector aumVec(maxIterations, -1.0);
+
+    lineSearch(
+            &lines[0],
+            lineCount,
+            &fpDiff[0],
+            &fnDiff[0],
+            initialAum,
+            maxIterations,
+            &FP[0], &FN[0], &M[0],
+            &stepSizeVec[0],
+            &aumVec[0]
+    );
+
+    return Rcpp::List::create(Rcpp::Named("aum", aumVec), Rcpp::Named("step.size", stepSizeVec));
```

Comment:
please return Rcpp::DataFrame instead of List since these two vectors are the same size